packages/hydrogen/src/cache/server-fetch.ts

```diff
@@ -60,6 +66,43 @@ export async function fetchWithServerCache<T = unknown>(
   return runWithCache(
     cacheKey,
     async () => {
+      if (streamConfig) {
```

Comment:
I think it's fine to branch when we detect `@defer` usage, because the underlying Shopify GraphQL client requires a branch. There's a separate `client.request` and `client.requestStream` implementation. And it throws if you use `defer` not with `client.requestStream`.

So instead of branching using either of their implementations, we branch, and use our original implementation for most requests, and only use theirs if it's defer. The implementation is slightly awkward because they don't expose the raw response object, which we need for caching and the sub-request profiler.